lib/rubocop/cop/style/safe_navigation.rb

```diff
@@ -51,18 +61,18 @@ class SafeNavigation < Cop
               (send $_ {:nil? :!}) nil
               {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
             ...)
-            (and
-              {(send (send $_ :nil?) :!) $_}
-              {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
-            ...)
-            (or
-              (send $_ {:nil? :!})
-              {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
-            ...)
           }
         PATTERN
 
+        def_node_matcher :candidate_that_may_introduce_nil, <<-PATTERN
+          (and
+            {(send (send $_ :nil?) :!) $_}
+            {(send $_ $_ ...) (block (send $_ $_ ...) ...)}
+          ...)
+        PATTERN
+
         def on_if(node)
+          return unless modifier_if?(node)
```

Comment:
I removed the check all together. I was able to get some time to finally figure out how to handle the auto correct properly.